TrackingProduction/Fun4All_TrackSeeding.C

```diff
@@ -0,0 +1,432 @@
+
+#include <kfparticle_sphenix/KFParticle_sPHENIX.h>
+
+#include <stdio.h>
+
+R__LOAD_LIBRARY(libkfparticle_sphenix.so)
+
+R__LOAD_LIBRARY(libfun4all.so)
+R__LOAD_LIBRARY(libffamodules.so)
+R__LOAD_LIBRARY(libphool.so)
+R__LOAD_LIBRARY(libcdbobjects.so)
+R__LOAD_LIBRARY(libTrackingDiagnostics.so)
+R__LOAD_LIBRARY(libtrackingqa.so)
+void Fun4All_TrackSeeding(
+    const int nEvents = 5,
+    const std::string clusterfilename = "DST_TRKR_CLUSTER_run2pp_ana441_2024p007-00052911-00000.root",
+    const std::string dir = "/sphenix/lustre01/sphnxpro/physics/slurp/tracking/ana441_2024p007/run_00052900_00053000/",
+    const std::string outfilename = "clusters_seeds",
+    const bool convertSeeds = false,
+    const bool doKFParticle = false)
+{
+  //std::string inputseedRawHitFile = dir + seedfilename;
+  std::string inputclusterRawHitFile = dir + clusterfilename;
+
+  G4TRACKING::convert_seeds_to_svtxtracks = convertSeeds;
+  std::cout << "Converting to seeds : " << G4TRACKING::convert_seeds_to_svtxtracks << std::endl;
+  std::pair<int, int>
+      runseg = Fun4AllUtils::GetRunSegment(clusterfilename);
+  int runnumber = runseg.first;
+  int segment = runseg.second;
+
+  auto rc = recoConsts::instance();
+  rc->set_IntFlag("RUNNUMBER", runnumber);
+
+  Enable::CDB = true;
+  rc->set_StringFlag("CDB_GLOBALTAG", "ProdA_2024");
+  rc->set_uint64Flag("TIMESTAMP", runnumber);
+  std::string geofile = CDBInterface::instance()->getUrl("Tracking_Geometry");
+
+  TpcReadoutInit(runnumber);
+  std::cout << " run: " << runnumber
+            << " samples: " << TRACKING::reco_tpc_maxtime_sample
+            << " pre: " << TRACKING::reco_tpc_time_presample
+            << " vdrift: " << G4TPC::tpc_drift_velocity_reco
+            << std::endl;
+
+  string outDir = "myKShortReco/";
+  string outputFileName = "outputKFParticle_" + to_string(runnumber) + "_" + to_string(segment) + ".root";
+  string outputRecoDir = outDir + "inReconstruction/";
+  string outputRecoFile = outputRecoDir + outputFileName;
+
+  if(doKFParticle){
+    string makeMainDirectory = "mkdir -p " + outDir;
+    system(makeMainDirectory.c_str());
+    string makeDirectory = "mkdir -p " + outputRecoDir;
+    system(makeDirectory.c_str());
+  }
+
+  // distortion calibration mode
+  /*
+   * set to true to enable residuals in the TPC with
+   * TPC clusters not participating to the ACTS track fit
+   */
+  G4TRACKING::SC_CALIBMODE = false;
+  TRACKING::pp_mode = true;
+
+  ACTSGEOM::mvtxMisalignment = 100;
+  ACTSGEOM::inttMisalignment = 100.;
+  ACTSGEOM::tpotMisalignment = 100.;
+  TString outfile = outfilename + "_" + runnumber + "-" + segment + ".root";
+  std::string theOutfile = outfile.Data();
+  auto se = Fun4AllServer::instance();
+  se->Verbosity(1);
+
+
+  Fun4AllRunNodeInputManager *ingeo = new Fun4AllRunNodeInputManager("GeoIn");
+  ingeo->AddFile(geofile);
+  se->registerInputManager(ingeo);
+
+  CDBInterface *cdb = CDBInterface::instance();
```

Comment:
We can remove this chunk from the macro, as I added it into tpc readout init so that it happens automatically each time.
 e.g. from line 118-130